engine/src/flutter/shell/platform/windows/host_window.h

```diff
@@ -85,27 +115,63 @@ class HostWindow {
     MONITORINFO monitor_info;
   };
 
-  HostWindow(WindowManager* controller,
+  HostWindow(WindowManager* window_manager,
              FlutterWindowsEngine* engine,
              WindowArchetype archetype,
-             std::unique_ptr<FlutterWindowsViewController> view_controller,
-             const BoxConstraints& constraints,
-             HWND hwnd);
+             DWORD window_style,
+             DWORD extended_window_style,
+             const BoxConstraints& box_constraints,
+             Rect const initial_window_rect,
+             LPCWSTR title,
+             HWND owner_window);
```

Comment:
Do you think we should make this a `std::optional<HWND>`?

I know HWND is a pointer itself that can be `nullptr`, but wrapping it in a `std::optional` would force people to remember to check for null. This might help reduce the likelihood of segfaults. What do you think?